### PR TITLE
Incorporate ShellCheck into OCBuild

### PR DESCRIPTION
**Pull request by dakanji**
Closes acidanthera/bugtracker/issues/#906
OCBuild currently has no CI in place

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+language: cpp
+env:
+  global:
+  - FORCE_INSTALL=1
```
Comment (vit9696) [inline .travis.yml:4]: You do not need `env` part at all, it is for `macbuild tool`. Are there any issues? If not, I can merge.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+matrix:
+  include:
+ - os: linux
```
Comment (vit9696) [inline .travis.yml:8]: You have a mistake here. Should be two spaces. I have already correct the same issue in OpenCorePkg:
https://github.com/acidanthera/OpenCorePkg/commit/1d68511a6a835b8b4005d90d0d16e0f0d899f496

Comment (dakanji) [reply .travis.yml:8]: Will fix

**Comment by dakanji** [comment]
Will address failing scripts that didn't come up before

**Comment by vit9696** [comment]
Note, that the version of shellcheck from Travis is old (0.6.0) and seems to be mishandling rc files. I downloaded the latest version manually here: https://github.com/acidanthera/OpenCorePkg/commit/ed841679e80cff774b2b61e2252d964b07ff4483.

However, I believe this code (and probably LLVM bootstrap below) needs to be changed to stop being duplicated. Please make a separate tool in ocbuild repository (and maybe LLVM if you do not mind) for shellcheck and bootstrap it similarly to how with deal with efibuild.sh or Lilu SDK. So that we end up having something like:

```
      - src=$(/usr/bin/curl -Lfs https://raw.githubusercontent.com/acidanthera/ocbuild/master/shellcheck.sh) && eval "$src" || exit 1
      # Check Parent Dir (Change "*.tool" to "*.{sh,tool}" as required if needed later)
      - shellcheck ./*.tool
      # Check Sub Dirs (Change "*.tool" to "*.{sh,tool}" as required if needed later)
      - shellcheck ./**/*.tool
```

**Comment by vit9696** [comment]
Also, maybe change your run command to something like this?

```
find . \( -name "*.tool" -o -name "*.sh" \) -exec shellcheck {} \;
```

**Comment by dakanji** [comment]
Homebrew always installs the latest version. Worth reverting to the Mac VM?

**Comment by vit9696** [comment]
To be honest, I do not mind either route, Mac VMs are slower, but it is not like we are in a hurry. If you want the Mac route, you do not need to set clang and xcode fields, can leave them as default as we are not compiling C code for that target.

**Comment by dakanji** [comment]
> ```
> find . \( -name "*.tool" -o -name "*.sh" \) -exec shellcheck {} \;
> ```
Seems a much better approach

**Comment by dakanji** [comment]
> To be honest, I do not mind either route, Mac VMs are slower, but it is not like we are in a hurry. If you want the Mac route, you do not need to set clang and xcode fields, can leave them as default as we are not compiling C code for that target.

Might be possible to install from pkg manager on Linux

**Comment by vit9696** [comment]
Package manager on ubuntu contains 0.3.9 version or so, and you installed it. Yet Travis version overrode it. I would say it is better to have something newer. To be honest, maybe use macOS vm indeed.

**Comment by dakanji** [comment]
> maybe use macOS vm

OK. Just checked and Homebrew is already on v0.7.1.
Will change the OpenCorePkg one first and work down the list

**Comment by dakanji** [comment]
Should the language here still be cpp or something else?

**Comment by dakanji** [comment]
Query:

ShellCheck has three relevant severity levels "error", "warning" and "info".
Default is  "warning" and this is when the CI Check will fail.

I have been manually fixing to address "info" and wonder whether you want the test to be set to this level or leave at the default "warning"?

"info" is for suggested improvements but not critical. Basically wondering how strict you want things to be

**Comment by vit9696** [comment]
Info sounds good to me as long as it is not too problematic.

**Comment by vit9696** [comment]
> Should the language here in travis.yml still be cpp or something else?

Yeah, it does not matter really.

**Pull request merged**